### PR TITLE
Sw 357 support dynamic groups

### PR DESCRIPTION
**Pull request by zaifastafa**

### CODE REVIEW

**Review by TheKeymaster — CHANGES_REQUESTED** [review]

**Review by TheKeymaster — CHANGES_REQUESTED** [review]

File: src/Export/DynamicProductGroupService.php [hunk 1]
```diff
@@ -119,93 +120,104 @@ public function setSalesChannel(SalesChannelEntity $salesChannelEntity): void
...
+        $products = [];
+        $cacheItem = $this->getCacheItem();
+        if ($cacheItem->get()) {
+            $products = unserialize($cacheItem->get());
         }
+        if (!Utils::isEmpty($products) && isset($products[$productId])) {
+            return $products[$productId];
+        }
+
+        return $products;
```
[the remark this answers is not in the document]
Comment (zaifastafa) [reply src/Export/DynamicProductGroupService.php:196]: Yeah my bad. I thought I was sending the empty array that I have initialized at the top of the method.

File: src/Export/DynamicProductGroupService.php [hunk 2]
```diff
@@ -119,93 +120,104 @@ public function setSalesChannel(SalesChannelEntity $salesChannelEntity): void
...
+    }
+
+    private function buildCriteria(): Criteria
```
[the remark this answers is not in the document]
Comment (zaifastafa) [reply src/Export/DynamicProductGroupService.php:199]: Sarcasm? 😜

[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Export/DynamicProductGroupService.php:202]: Not at all! Really nice abstraction. Please more of that in the future!

File: src/Export/DynamicProductGroupService.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Export/DynamicProductGroupService.php]: No worries, we all make mistakes! Reading the code I was like, what?!

File: src/Export/DynamicProductGroupService.php [hunk 3]
```diff
@@ -119,93 +120,104 @@ public function setSalesChannel(SalesChannelEntity $salesChannelEntity): void
...
     /**
      * @return CategoryEntity[]
      */
     public function getCategories(string $productId): array
     {
-        if (!Utils::isEmpty($this->products) && isset($this->products[$productId])) {
-            return $this->products[$productId];
+        $products = [];
+        $cacheItem = $this->getCacheItem();
+        if ($cacheItem->get()) {
```
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Export/DynamicProductGroupService.php]: Ah yes, it was called `isHit`.

**Review by TheKeymaster — APPROVED** [review]

**Review by TheKeymaster — APPROVED** [review]

**Pull request merged**